appveyor-tool.ps1

```diff
@@ -0,0 +1,119 @@
+    Progress ("Downloading NETCFv35PowerToys from: " + $url)
+    Invoke-WebRequest -Uri $url -OutFile NETCFv35PowerToys.msi
+    
+    Progress("Running NETCFSetupv35 installer")
+  
+    $msi = @("NETCFSetupv35.msi","NETCFv35PowerToys.msi")
+    foreach ($msifile in $msi) 
+    {
+    if(!(Test-Path($msi)))
+    {
+        throw "MSI files are not present, please check logs."
+    }
+    Progress("Installing msi " + $msifile )
+    Start-Process -FilePath "$env:systemroot\system32\msiexec.exe" -ArgumentList "/i `"$msifile`" /qn /norestart" -Wait -WorkingDirectory $pwd  -RedirectStandardOutput stdout.txt -RedirectStandardError stderr.txt
+    $OutputText = get-content stdout.txt
+    Progress($OutputText)
+    $OutputText = get-content stderr.txt
+    Progress($OutputText) 
+    }
+    if(!(Test-Path("C:\Windows\Microsoft.NET\Framework\v3.5\Microsoft.CompactFramework.CSharp.targets")))
+    {
+        throw "Compact framework files not found after install, install may have failed, please check logs."
+    }
+    RegistryWorkAround
+}
+
+Function Progress
+{
+    [CmdletBinding()]
+    param (
+        [Parameter(Position=0, Mandatory=0)]
+        [string]$Message = ""
+    )
+    $ProgressMessage = '== ' + (Get-Date) + ': ' + $Message
+
+    Write-Host $ProgressMessage -ForegroundColor Magenta
+}
+
+function Test-Administrator  
+{  
+    $user = [Security.Principal.WindowsIdentity]::GetCurrent();
+    (New-Object Security.Principal.WindowsPrincipal $user).IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator)  
+}
+
+function RegistryWorkAround
+{
+    ## http://community.sharpdevelop.net/forums/t/10548.aspx
+    ## see above link for work around for error 
+    ## The "AddHighDPIResource" task failed unexpectedly.
+    ## System.ArgumentNullException: Value cannot be null.
+    ## Parameter name: path1 
+    $registryPaths = @("HKLM:\SOFTWARE\Microsoft\VisualStudio\9.0\Setup\VS","HKLM:\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\9.0\Setup\VS")
+    ## not last reply in forum post about needing the entry in WOW is why there is two paths
+    $Name = "ProductDir"
+    $value = "C:\Program Files (x86)\Microsoft Visual Studio 9.0"
+    
+    foreach($registryPath in $registryPaths)
+    {
+        If(!(Test-Path $registryPath))
+        {
+            New-Item -Path $registryPath -Force | Out-Null
+        }
+        If(!(Test-Path $registryPath+"\"+$Name))
+        {
+        New-ItemProperty -Path $registryPath -Name $name -Value $value `
+            -PropertyType String -Force | Out-Null
+        }
+        If(!((Get-ItemProperty -Path $registryPath -Name $Name).ProductDir -eq "C:\Program Files (x86)\Microsoft Visual Studio 9.0"))
+        {
+            throw "Registry path " + $registryPath + " not set to correct value, please check logs"
+        }
+        else
+        {
+            Progress("Regsitrty update ok to value " + (Get-ItemProperty -Path $registryPath -Name $Name).ProductDir)
```

Comment:
yes!... i haven't been able to find a spell checker for VS Code that will check the spelling in my comments and strings, the only one available only appears to do markdown files